tools/dissector/sample_manager.cpp

```diff
@@ -36,211 +39,218 @@ namespace OpenDDS
+          }
+          else {
+            dissector->add_field(new Sample_Field(Sample_Field::Char, ""));
+          }
+          break;
+        case 16:
+          if (i.isUnsigned()) {
+            dissector->add_field(new Sample_Field(Sample_Field::UShort, ""));
+          }
+          else {
+            dissector->add_field(new Sample_Field(Sample_Field::Short, ""));
+          }
+          break;
+        case 32:
+          if (i.isUnsigned()) {
+            dissector->add_field(new Sample_Field(Sample_Field::ULong, ""));
+          }
+          else {
+            dissector->add_field(new Sample_Field(Sample_Field::Long, ""));
+          }
+          break;
+        case 64:
+          if (i.isUnsigned()) {
+            dissector->add_field(new Sample_Field(Sample_Field::ULongLong, ""));
+          }
+          else {
+            dissector->add_field(new Sample_Field(Sample_Field::LongLong, ""));
+          }
+          break;
+        default:
+          {
+            ACE_DEBUG ((LM_WARNING, ACE_TEXT ("Unknown integer type\n")));
+
+            const unsigned int seven = 7;
+            const unsigned int bytes = ((i.bits() + seven) & (~seven)) / 8;
+            for (unsigned int i = 0; i != bytes; ++i) {
+              dissector->add_field(new Sample_Field(Sample_Field::Octet, ""));
+            }
           }
+          break;
         }
+      }
 
-        void visit (itl::Fixed&) {
-          dissector = new Sample_Dissector();
+      void visit (itl::Float& f) {
+        dissector = new Sample_Dissector();
+        switch (f.model()) {
+        case itl::Float::Binary32:
+          dissector->add_field(new Sample_Field(Sample_Field::Float, ""));
+          break;
+        case itl::Float::Binary64:
+          dissector->add_field(new Sample_Field(Sample_Field::Double, ""));
+          break;
+        case itl::Float::Binary128:
+          dissector->add_field(new Sample_Field(Sample_Field::LongDouble, ""));
+          break;
+        default:
           dissector->add_field(new Sample_Field(Sample_Field::Undefined, ""));
-          ACE_DEBUG ((LM_WARNING, ACE_TEXT ("Fixed-point types are not supported\n")));
+          ACE_DEBUG ((LM_WARNING, ACE_TEXT ("Unknown float model: %d\n"), f.model()));
+          break;
         }
+      }
 
-        Sample_Dissector* do_array(std::vector<unsigned int>::const_iterator pos,
-                                   std::vector<unsigned int>::const_iterator limit,
-                                   itl::Type* element_type) {
-          if (pos != limit) {
-            return new Sample_Array(*pos, do_array(pos + 1, limit, element_type));
-          }
-          else {
-            return get_dissector(map, element_type);
-          }
+      void visit (itl::Fixed&) {
+        dissector = new Sample_Dissector();
+        dissector->add_field(new Sample_Field(Sample_Field::Undefined, ""));
+        ACE_DEBUG ((LM_WARNING, ACE_TEXT ("Fixed-point types are not supported\n")));
```

Comment:
OpenDDS does support the `fixed` data type now (probably not when this code was first written) for some use cases.  If you have time it would be good to add it.